### ISSUE TITLE
Array Element By Position

### ISSUE THREAD

**Issue by zenom** [hmarr/mongoengine#36]
Array Element by Position

Array elements also may be accessed by specific array position:

// i.e. comments[0].by == "Abe"

> db.blogposts.find( { "comments.0.by" : "Abe" } )
> (The above examples use the mongo shell's Javascript syntax.  The same operations can be done in any language for which Mongo has a driver available.)

Would be nice to have this sort of syntax as well. I recently had the need to do something like the following:

db.program.update({"program": "test", "sites.slug": "site-1"}, {$set: {"sites.$.disabled": true}})

**Comment by flosch** [comment]
Don't get it. This still works:

```
class Comment(Document):
    by = StringField()

class Post(Document):
    comments = ListField(ReferenceField(Comment))

c1 = Comment.objects.create(by='abe')
c2 = Comment.objects.create(by='abe')
c3 = Comment.objects.create(by='abe')

p = Post.objects.create(comments=[c1, c2, c3])

print p.comments[0]
```

Do you have an example in MongoEngine where ME lacks your desired feature?

**Comment by zenom** [comment]
This ticket was 3 mos ago, I don't really remember what the deal was other than it was all embedded documents and trying to edit/delete them.  If it works then feel free to close this.

**Comment by flosch** [comment]
Okay. Just tested it with EmbeddedDocuments, still works. I suggest to close this issue.

**Comment by hmarr** [comment]
Added support for querying by array position. Closed by 2af5f3c56ebc7feff6a0a1cd95a77e12b425efed.
